### PR TITLE
reuse webdav client for dav

### PR DESCRIPTION
**Pull request by vauvenal5**
So basically this is my idea how to reuse the webdav client for dav. Not sure if I am missing something or not. Let me know if you had something different in mind.

### CODE REVIEW

File: packages/nextcloud/lib/src/webdav/path_uri.dart [hunk 1]
```diff
@@ -12,6 +12,12 @@ class PathUri {
     required this.pathSegments,
   });
 
+  /// Base path used for dav on the server
+  factory PathUri.dav() => PathUri.parse('/remote.php/dav');
+
+  /// Base path used for webdav on the server
+  factory PathUri.webdav() => PathUri.parse('/remote.php/webdav');
+
```
Comment (Leptopoda) [inline packages/nextcloud/lib/src/webdav/path_uri.dart:20]: The `PathUri` should be a `Uri` like class and not provide any paths itself.

Maybe create an enum with a getter that returns the uri.

File: packages/nextcloud/lib/webdav.dart
Comment (Leptopoda) [inline packages/nextcloud/lib/webdav.dart:1]: The extension from `packages/nextcloud/lib/dav.dart` should also be moved into here.
The other one is not going to be removed automatically if the spec isn't there anymore.

**Review by Leptopoda — COMMENTED** [review]
Some minor code style feedback for now.
tests are obviously still needed and I don't pretend to know the difference between the two dav endpoints. I also do not understand why we should remove the specs for the one.
@provokateurin can propably review this side better.

File: packages/nextcloud/lib/src/webdav/file.dart [hunk 2]
```diff
@@ -28,7 +27,7 @@ class WebDavFile {
     return PathUri(
       isAbsolute: false,
       isDirectory: href.isDirectory,
-      pathSegments: href.pathSegments.sublist(webdavBase.pathSegments.length),
+      pathSegments: href.pathSegments.sublist(PathUri.webdav().pathSegments.length),
```
Comment (provokateurin) [inline packages/nextcloud/lib/src/webdav/file.dart:30]: Shouldn't this still use the base?

File: packages/nextcloud/lib/src/api/dav.openapi.json
Comment (provokateurin) [inline packages/nextcloud/lib/src/api/dav.openapi.json:1]: These files should not be removed.

File: packages/nextcloud/lib/dav.dart
Comment (provokateurin) [inline packages/nextcloud/lib/dav.dart:1]: This file should also not be changed.

File: packages/nextcloud/lib/src/webdav/path_uri.dart (hunk 1 above)
Comment (provokateurin) [reply packages/nextcloud/lib/src/webdav/path_uri.dart:20]: IMO this can just be hardcoded in the constructors of the WebDavClient.

**Review by provokateurin — CHANGES_REQUESTED** [review]
I'm not sure if this approach is the best.
The naming conflicts are definitely annoying here :/

File: packages/nextcloud/lib/dav.dart
Comment (vauvenal5) [reply packages/nextcloud/lib/dav.dart:1]: I believe I am misunderstanding something. What is the difference between the dav client and the webdav client? To my current understanding the generated clients are basically for OpenAPI endpoints. So why is for dav a client generated but not for webdav?

Comment (provokateurin) [reply packages/nextcloud/lib/dav.dart:1]: The generated client is from OpenAPI, but it is only for some REST endpoints of the [`dav`](https://github.com/nextcloud/server/tree/master/apps/dav) app.

File: packages/nextcloud/lib/src/webdav/file.dart (hunk 2 above)
Comment (vauvenal5) [reply packages/nextcloud/lib/src/webdav/file.dart:30]: I was looking into that too, it is not ideal, however getting the actual base path there is not trivial due to the XML handling. I will be thinking about this one more time but wanted general feedback first.

**Comment by provokateurin** [comment]
@vauvenal5 please make sure that everything is covered with tests, otherwise we will have a hard time verifying that it works as intended.

**Comment by vauvenal5** [comment]
I will close this one and re-open with the new changes latter on.

**Pull request closed without merging**